indexers/dbms/PostgreSQLIndexer/__init__.py

```diff
@@ -0,0 +1,158 @@
+    :param collection: the collection name
+    :param args: other arguments
+    :param kwargs: other keyword arguments
+    """
+
+    def __init__(
+        self,
+        hostname: str = "127.0.0.1",
+        port: int = 5432,
+        username: str = "postgres",
+        password: str = "123456",
+        database: str = "postgres",
+        table: Optional[str] = "default_table",
+        *args,
+        **kwargs
+    ):
+
+        super().__init__(*args, **kwargs)
+
+        self.hostname = hostname
+        self.port = port
+        self.username = username
+        self.password = password
+        self.database_name = database
+        self.table = table
+        self.connect()
+
+    def get_connection(self):
+        return self.connect()
+
+    def connect(self):
+        """Connect to the database. """
+
+        import psycopg2
+        from psycopg2 import Error
+
+        try:
+            self.connection = psycopg2.connect(
+                user=self.username,
+                password=self.password,
+                database=self.database_name,
+                host=self.hostname,
+                port=self.port,
+            )
+            self.cursor = self.connection.cursor()
+            self.logger.info("Successfully connected to the database")
+            self.create_table()
+            self.connection.commit()
+        except (Exception, Error) as error:
+            self.logger.error("Error while connecting to PostgreSQL", error)
+        return self
+
+    def create_table(self):
+        """
+        Create Table with id, vecs and metas.
+        """
+
+        self.cursor.execute("select exists(select * from information_schema.tables where table_name=%s)", (self.table,))
+        if self.cursor.fetchone()[0]:
+            self.logger.info("Using existing table")
+        else:
+            try:
+                self.cursor.execute(
+                    f"""DROP TABLE IF EXISTS {self.table};
+                                    CREATE TABLE {self.table} (
+                                    ID INT PRIMARY KEY, 
+                                    VECS BYTEA, 
+                                    METAS BYTEA);"""
+                )
+                self.logger.info("Successfully table created")
+            except:
+                self.logger.error("Error while creating table!")
+
+    def add(self, ids, vecs, metas, *args, **kwargs):
+        """Insert the documents into the database.
+
+        :param ids: List of doc ids to be added
+        :param vecs: List of vecs to be added
+        :param metas: List of metas of docs to be added
+        :return record: List of Document's id added
+        """
+
+        self.cursor.execute(f"DELETE FROM {self.table}")
+        for i in range(len(ids)):
+            self.cursor.execute(
+                f"INSERT INTO {self.table} (ID, VECS, METAS) VALUES (%s, %s, %s)",
+                (ids[i], pickle.dumps(vecs), pickle.dumps(metas)),
```

Comment:
we dump all vecs and metas for each id, is this what we want? 
Or should we use `pickle.dumps(vecs[i])`

Reply:
Separated, yes. It might be used as KV retrieval if people don't want to start another KV on the query side